source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst

```diff
@@ -0,0 +1,95 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_set_matrix_property:
+
+set_matrix_property
+===================
+
+Matrix properties
+-----------------
+
+.. container:: section
+
+   .. code:: cpp
+
+      enum class matrix_property {
+          symmetric,
+          sorted,
+      };
+
+   TODO: Clarify are the properties meant to be used as a bit mask so multiple
+   properties can be set?
+
+   .. list-table::
+      :header-rows: 1
+      :widths: 20 80
+
+      * - Value
+        - Description
+      * - ``symmetric``
+        - Guarantees that the matrix is symmetric, meaning it must be square and
+          its transpose is equal to itself. TODO: Confirm whether the user can
+          only provide half of the data in this case?
```

Comment:
@gajanan-choudhary, I could use your advice to improve this description for what you were planning exactly.

Reply:
I'd say "Guarantees that the user-provided matrix data are symmetric, meaning the matrix is square, the user data contain both lower and upper triangular regions, and that its transpose is equal to itself."

Users do not provide half the data in such a case; the users provide the full data and are indicating that both lower and upper regions are stored, and that the sparsity pattern and values are both symmetric.

"Properties" are specific to the data the users provide to us. If they say it is sorted, it's user's responsibility to ensure they are sorted. If they are symmetric, it's their responsibility to ensure that their arrays follow that.

Reply:
Thanks for the good suggestion, done in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/094292667371ff600d778e87f9d0f69055ba0268